app/Transformers/BeatmapsetTransformer.php

```diff
@@ -49,6 +50,7 @@ public function transform(Beatmapset $beatmapset = null)
             'artist' => $beatmapset->artist,
             'play_count' => $beatmapset->play_count,
             'favourite_count' => $beatmapset->favourite_count,
+            'has_favorited' => $beatmapset->hasFavorited(Auth::user()),
```

Comment:
Looks like this requires some kind of preloading on Beatmapsets' search/index otherwise it'll loop n times just to get this for each sets.